core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java

```diff
@@ -220,4 +250,18 @@ private boolean canLogAtLevel(int level) {
 
         return args;
     }
+
+    /**
+     * ClientLogger that doesn't perform any logging.
+     */
+    static final class NoopClientLogger extends ClientLogger {
```

Comment:
We likely don't need this given that the ClientLogger will no-op if the log level isn't supported.